use-cases/bed-allocation/src/main/java/org/acme/bedallocation/domain/Schedule.java

```diff
@@ -0,0 +1,126 @@
+package org.acme.bedallocation.domain;
+
+import java.util.List;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningEntityCollectionProperty;
+import ai.timefold.solver.core.api.domain.solution.PlanningScore;
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.api.domain.solution.ProblemFactCollectionProperty;
+import ai.timefold.solver.core.api.domain.valuerange.ValueRangeProvider;
+import ai.timefold.solver.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
+import ai.timefold.solver.core.api.solver.SolverStatus;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@PlanningSolution
+public class Schedule {
+
+    private List<Department> departments;
+
+    @PlanningEntityCollectionProperty
+    private List<BedDesignation> bedDesignations;
+
+    @PlanningScore
+    private HardMediumSoftScore score;
+
+    private SolverStatus solverStatus;
+
+    // No-arg constructor required for Timefold
+    public Schedule() {
+    }
+
+    public Schedule(HardMediumSoftScore score, SolverStatus solverStatus) {
+        this.score = score;
+        this.solverStatus = solverStatus;
+    }
+
+    // ************************************************************************
+    // Getters and setters
+    // ************************************************************************
+    @ProblemFactCollectionProperty
+    public List<Department> getDepartments() {
+        return departments;
+    }
+
+    public void setDepartments(List<Department> departments) {
+        this.departments = departments;
+    }
+
+    @JsonIgnore
+    @ProblemFactCollectionProperty
+    public List<DepartmentSpecialism> getDepartmentSpecialisms() {
+        return departments.stream()
+                .flatMap(d -> d.getSpecialismsToPriority().entrySet().stream()
+                        .map(e -> new DepartmentSpecialism("%s-%s".formatted(d.getId(), e.getKey()), d, e.getKey(),
+                                e.getValue()))
+                        .toList()
+                        .stream())
+                .toList();
```

Comment:
I'd be very careful about this pattern.

First of all - this will do some very heavy lifting every time it is called.
But more importantly, this means that every time it is called, the facts will be different instances of objects, and that will eventually lead to inefficiencies.

Personally, I'd fill these fields when the departmets are being set, and not touch them after.